spec/did_you_know_spec.rb

```diff
@@ -0,0 +1,68 @@
+RSpec.describe Geordi::Hint do
+
+  subject { Geordi::Hint }
+
+  describe '#did_you_know' do
+
+    it 'prints a random hint in the console if an array with suitable suggestions was provided and returns all generated hints' do
+      allow(Geordi::Hint).to receive(:hint).with(:branch).and_return('Did you know? `geordi branch` can check out a feature branch based on a story from Pivotal Tracker')
+      allow(Geordi::Hint).to receive(:hint).with([:branch, :from_master]).and_return('Did you know? `geordi branch -m` can branch from master instead of the current branch')
+      allow(Geordi::Hint).to receive(:hint).with('`geordi branch`').and_return('Did you know? `geordi branch`')
```

Comment:
You're stubbing the hint generation here. Why? The hints are the most important part of this method. I suggest to not stub them, but make them part of the test instead. In `describe '#hint'` below, you only need to care about edge cases then. (As I suggested above, #hint should probably be a private method, which we avoid to test.)

---
Generally on stubbing: you should avoid using real input and output when stubbing. The test _looks_ like normal, but all values were stubbed. When stubbing, always use made-up, self-explaining values like this:
```suggestion
      allow(Geordi::Hint).to receive(:hint).with(:command_1).and_return('Did you know 1')
      allow(Geordi::Hint).to receive(:hint).with(:command_2).and_return('Did you know 2')
```
Then in each part of your test, it gets immediately clear you're working with example data.